pisa/core/param.py

```diff
@@ -839,7 +839,7 @@ def __setattr__(self, attr, val):
             if isinstance(val, Param):
                 assert val.name == attr
                 self._params[idx] = val
-            elif isbarenumeric(val):
+            elif isinstance(val, ureg.Quantity):
```

Comment:
Why did you change this? (Just making sure you are fixing a bug, and not introducing a bug for something you haven't considered.)

Reply:
This is a bug fix. The values of Param objects are quantities with units, rather than pure numeric values.

Reply:
I just checked out your code and ran the unit tests. This appears to break the use case (which previously worked) of:

```my_param_set.x = 1```

although I agree it now allows

```my_param_set.x = ureg.Quantity("1 meter")```

Note that there are a lot of unit tests, especially in the core modules. Simply run the file to run the unit tests, and you'll see that you broke one here.

I fixed the bug you have introduced, added a unit test to ensure that the second use case you introduced gets tested from now on, and I will submit a PR against _your_ branch of the code. If the changes I made are okay, then you can merge my PR into your branch and the changes should show up here.

Reply:
Ah ok, thank you!